File: TypeScript/Arrays/Search/Interpolation.ts
const Interpolation = (arr: number[], target: number): number => {
  let low: number = 0;
  let high: number = arr.length - 1;
  while (low <= high) {
    let pos: number =
      low +
      Math.floor(((high - low) * (target - arr[low])) / (arr[high] - arr[low]));

    if (arr[pos] == target) {
      return pos;
    }
    if (arr[pos] < target) {
      low = pos + 1;
    } else {
      high = pos - 1;
    }
  }
  return -1;
};

let arrS: number[] = [1, 2, 3, 4, 5, 6, 7, 8, 9];
console.log(Interpolation(arrS, 7));

File: Python/Arrays/Search/Interpolation.py
def Interpolation(arr:list[int], target:int) -> int:
    low : int = 0
    high: int = len(arr) -1
    while low <= high:
        pos:int = low + ((high -low) * (target - arr[low]) // (arr[high] - arr[low]))
        if arr[pos] == target:
            return pos
        if arr[pos] > target:
            high = pos - 1
        else:
            low = pos + 1
    return -1


arr:list[int] = [12,24,35,46,57,68,79]
print(Interpolation(arr, 12))